endpoints/setuid.go

```diff
@@ -76,7 +76,7 @@ func NewSetUIDEndpoint(cfg config.HostCookie, perms gdpr.Permissions, pbsanalyti
 			so.Success = true
 		}
 
-		pc.SetCookieOnResponse(w, cfg.Domain, cookieTTL)
+		pc.SetCookieOnResponse(w, r, cfg.Domain, cookieTTL)
```

Comment:
Isn't it better to perform the browser name and browser version "stripping" operation back in the endpoint itself? We can probably avoid passing the `http.Request` pointer to `SetCookieOnResponse` and the expensive dereference operation we have to do on it in favor of a single boolean value that signals the difference in the cookie we set. Summarizing the possible scenarios that the logic added into `SetCookieOnResponse` gives us:
```
   | strings.Index(ua,"Chrome/") | strings.Index(ua,"CriOS/") | version>=chromeMinVer ||               result
--------------------------------------------------------------------------------------------------------------------------------------
 1 |        FALSE                |          FALSE             |         N/A           || w.Header().Add("Set-Cookie", cookieStr) //"cookieStr" without the "SameSiteAttribute" part
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 2 |        FALSE                |          FALSE             |         N/A           || w.Header().Add("Set-Cookie", cookieStr) //"cookieStr" without the "SameSiteAttribute" part
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 3 |        FALSE                |          TRUE              |       FALSE           || w.Header().Add("Set-Cookie", cookieStr) //"cookieStr" without the "SameSiteAttribute" part
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 4 |        FALSE                |          TRUE              |       TRUE            || w.Header().Add("Set-Cookie", cookieStr+SameSiteAttribute)
   |                             |                            |                       || w.Header().Add("Set-Cookie", sameSiteCookieStr)
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 5 |        TRUE                 |          FALSE             |       FALSE           || w.Header().Add("Set-Cookie", cookieStr) //"cookieStr" without the "SameSiteAttribute" part
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 6 |        TRUE                 |          FALSE             |       TRUE            || w.Header().Add("Set-Cookie", cookieStr+SameSiteAttribute)
   |                             |                            |                       || w.Header().Add("Set-Cookie", sameSiteCookieStr)
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 7 |        TRUE                 |          TRUE              |       FALSE           || w.Header().Add("Set-Cookie", cookieStr) //"cookieStr" without the "SameSiteAttribute" part
---|-----------------------------|----------------------------------------------------||----------------------------------------------
 8 |        TRUE                 |          TRUE              |       TRUE            || w.Header().Add("Set-Cookie", cookieStr+SameSiteAttribute)
   |                             |                            |                       || w.Header().Add("Set-Cookie", sameSiteCookieStr)
--------------------------------------------------------------------------------------------------------------------------------------
```

We can probably implement `func chromeCheck(ua *UserAgent) (isChrome bool, isChromeOSS bool, isOverMinVersion bool)` inside `endpoints/setuid.go` and evaluate the returning values into an statement like: `var setSiteCookie bool = (isChrome && isOverMinVersion) || (isChromeOSS && isOverMinVersion)` so we can a call a version of `SetCookieOnResponse` that takes a boolean value instead of a pointer to the `http.Request` like:
```
 17 func NewSetUIDEndpoint(cfg config.HostCookie, perms gdpr.Permissions, pbsanalytics analytics.PBSAnalyticsModule, metrics pbsmetrics.MetricsEngine) httprouter.Handle {
 18     cookieTTL := time.Duration(cfg.TTL) * 24 * time.Hour
 19     return httprouter.Handle(func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 20         so := analytics.SetUIDObject{
 21             Status: http.StatusOK,
 22             Errors: make([]error, 0),
 23         }
 24 +-- 45 lines: defer pbsanalytics.LogSetUIDObject(&so)-----------------------------------------------------------------------------------------------------------------
 69
 70         if err == nil {
 71             labels := pbsmetrics.UserLabels{
 72                 Action: pbsmetrics.RequestActionSet,
 73                 Bidder: openrtb_ext.BidderName(bidder),
 74             }
 75             metrics.RecordUserIDSet(labels)
 76             so.Success = true
 77         }
 78         
      +     isChrome , isChromeOSS , isOverMinVersion := chromeCheck(ua *UserAgent)
      +     var setSiteCookie bool = (isChrome && isOverMinVersion) || (isChromeOSS && isOverMinVersion)
      +  
      -     pc.SetCookieOnResponse(w, r, cfg.Domain, cookieTTL)
 79   +     pc.SetCookieOnResponse(w, setSiteCookie, cfg.Domain, cookieTTL)
 80
 81     })
 82 }
endpoints/setuid.go
```
Or even better, just make:
```
      setSiteCookie := siteCookieCheck(ua *UserAgent)
```

Reply:
Yes, it's better to do the browser version check at the endpoint level. I've moved the code that checks for browser version to `endpoints/setuid.go`.